frontend/pause.py

```diff
@@ -0,0 +1,141 @@
+        self.retry = Button(self.screen, self.retry_image, self.x_percent(556), self.y_percent(619), self.scale_factor)
+        self.exit = Button(self.screen, self.back_image, self.x_percent(556 + 335/2), self.y_percent(619), self.scale_factor)
+
+        self.hide = True
+        self.p_key_was_pressed = False
+
+        self.hide = True
+        self.p_key_was_pressed = False
+
+    def load_assets(self):
+        """
+        Loads necessary assets.
+        """
+        # load asset paths then images
+        background_path = os.path.join(ASSETS_DIRECTORY, "background.png")
+        title_path = os.path.join(ASSETS_DIRECTORY, "title.png")
+        bar_foreground_path = os.path.join(ASSETS_DIRECTORY, "bar_foreground.png")
+        bar_background_path = os.path.join(ASSETS_DIRECTORY, "bar_background.png")
+        retry_path = os.path.join(ASSETS_DIRECTORY, "replay-button.png")
+        back_path = os.path.join(ASSETS_DIRECTORY, "exit-button.png")
+        resume_path = os.path.join(ASSETS_DIRECTORY, "play.png")
+        minus_path = os.path.join(ASSETS_DIRECTORY, "minus.png")
+        plus_path = os.path.join(ASSETS_DIRECTORY, "plus.png")
+
+        self.background_image = LoadingScreen.ASSET[background_path]
+        self.title_image = LoadingScreen.ASSET[title_path]
+        self.bar_fg_image = LoadingScreen.ASSET[bar_foreground_path]
+        self.bar_bg_image = LoadingScreen.ASSET[bar_background_path]
+
+        self.retry_image = LoadingScreen.ASSET[retry_path]
+        self.back_image = LoadingScreen.ASSET[back_path]
+        self.resume_image = LoadingScreen.ASSET[resume_path]
+        self.minus_image = LoadingScreen.ASSET[minus_path]
+        self.plus_image = LoadingScreen.ASSET[plus_path]
+
+    
+    def draw(self):
+        """Draws all the screen components."""
+        if not self.hide:
+            self.background.draw()
+            self.title.draw()
+            self.pause_title.draw()
+            
+            self.music_title.draw()
+            self.sfx_title.draw()
+            self.music_slider.draw()
+            self.music_plus.draw()
+            self.music_minus.draw()
+            self.sfx_slider.draw()
+            self.sfx_plus.draw()
+            self.sfx_minus.draw()
+            self.resume.draw()
+            self.retry.draw()
+            self.exit.draw()
+
+        else:
+            self.screen.fill((0, 0, 0, 0))
+
+    def toggle(self):
+        """
+        Toggles the visibility of the pause screen.
+        """
+        self.hide = not self.hide
+
+    def is_active(self):
+        """
+        Returns whether the pause screen is currently active (visible).
+        """
+        return not self.hide
+    
+    def update(self, events):
+        """Update the screen and handle events."""
+        self.draw()
+
+        if self.hide:
+            return
+        
+        for event in events:
+            if self.exit.handle_event(event):
```

Comment:
add some comments for each conditionals so it is immediately clear what is going on here e.g. `# Return to main menu when exit is pressed`. for example, I don't immediately know what `handle_event` does. can we also rename some of these variables so its clearer what they are. `self.exit_button.handle_event(event)` makes it clearer what kind of events are being handled.